File: test-suite-java/src/test/java/io/micronaut/sourcegen/example/MyRepository1Test.java
/*
 * Copyright 2003-2021 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.sourcegen.example;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Optional;

class MyRepository1Test {
    @Test
    public void test() throws Exception {
        MyRepository1Instance myInstance = new MyRepository1Instance();
        MyEntity1 myEntity1 = new MyEntity1();

        Assertions.assertInstanceOf(CrudRepository1.class, myInstance);

        Assertions.assertTrue(myInstance.findById(123L).isEmpty());

        myInstance.save(myEntity1);

        Assertions.assertEquals(myEntity1, myInstance.findById(123L).get());
        Assertions.assertEquals(myEntity1, myInstance.findAll().iterator().next());
    }

    static class MyRepository1Instance implements MyRepository1 {

        MyEntity1 myEntity1;

        @Override
        public Optional<MyEntity1> findById(Long id) {
            return Optional.ofNullable(myEntity1);
        }

        @Override
        public List<MyEntity1> findAll() {
            return List.of(myEntity1);
        }

        @Override
        public void save(MyEntity1 entity) {
            myEntity1 = entity;
        }
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/sourcegen/example/MyRepository1Test.kt
package io.micronaut.sourcegen.example

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.*


internal class MyRepository1Test {
    @Test
    @Throws(Exception::class)
    fun test() {
        val myInstance = MyRepository1Instance()
        val myEntity1 = MyEntity1()
        Assertions.assertTrue(myInstance is CrudRepository1<*, *>)
        Assertions.assertTrue(myInstance.findById(123L).isEmpty())
        myInstance.save(myEntity1)
        Assertions.assertEquals(myEntity1, myInstance.findById(123L).get())
        Assertions.assertEquals(myEntity1, myInstance.findAll().iterator().next())
    }

    internal class MyRepository1Instance : MyRepository1 {
        var myEntity1: MyEntity1? = null
        override fun findById(id: Long): Optional<MyEntity1> {
            return Optional.ofNullable<MyEntity1>(myEntity1)
        }

        override fun findAll(): List<MyEntity1> {
            return java.util.List.of<MyEntity1>(myEntity1)
        }

        override fun save(entity: MyEntity1) {
            myEntity1 = entity
        }
    }
}
